src/utils/evm/test-routes.ts

```diff
@@ -0,0 +1,339 @@
+
+import { EVMAssetTransfer, EVMGenericMessageTransfer, Environment, EthereumConfig, EvmResource, ResourceType, SubstrateConfig } from '@buildwithsygma/sygma-sdk-core'
+import { BigNumber, Wallet, providers, utils } from 'ethers'
+import { RpcEndpoints } from '../../types'
+
+import { KeyringPair } from '@polkadot/keyring/types'
+import { Storage__factory } from '../Contracts'
+import { ApiPromise, WsProvider } from '@polkadot/api'
+import { AccountInfo } from '@polkadot/types/interfaces'
+import { LoggingData, fetchTokenAmount, sleep, waitUntilBridgedFungibleEvm } from '../helpers'
+import { print } from 'gluegun'
+
+const executionContractAddresses = {
+  EXECUTE_CONTRACT_ADDRESS_1:
+    '0xdFA5621F95675D37248bAc9e536Aab4D86766663',
+  EXECUTE_CONTRACT_ADDRESS_2:
+    '0x0e963aEe445EDC19034e9938570E5E7BE4Ee19Cd'
+}
+const EXECUTE_FUNCTION_SIGNATURE = '0xa271ced2'
+const MAX_FEE = '3000000'
+export async function testEvmRoutes(
+  ethereumConfigs: Array<EthereumConfig>,
+  SubstrateConfig: Array<SubstrateConfig>,
+  rpcEndpoints: RpcEndpoints,
+  evmWallet: Wallet,
+  substrateWallet: KeyringPair,
+  environment: Environment,
+): Promise<string> {
+  let result = ""
+  for (const network of ethereumConfigs) {
+    const sourceProvider = new providers.JsonRpcProvider(rpcEndpoints[network.chainId])
+    evmWallet = new Wallet(evmWallet.privateKey, sourceProvider)
+    const functionCalls = [] as any
+
+    // EVM -> EVM transfer
+    for (const resource of network.resources) {
+      for (const destinationDomain of ethereumConfigs) {
+        const loggingData = {
+          resourceId: resource.resourceId,
+          sourceDomainId: network.id,
+          sourceDomainName: network.name,
+          destinationDomainId: destinationDomain.id,
+          destinationDomainName: destinationDomain.name
+        }
+        try {
+          const destinationProvider = new providers.JsonRpcProvider(rpcEndpoints[destinationDomain.chainId])
+
+          if (destinationDomain.id == network.id) {
+            continue
+          }
+
+          for (const destinationResource of destinationDomain.resources as unknown as Array<EvmResource>) {
+            if (destinationResource.resourceId === resource.resourceId) {
+              switch (resource.type) {
+                case ResourceType.FUNGIBLE:
+                  const assetTransfer = new EVMAssetTransfer()
```

Comment:
Would be a bit nicer if it is a separate function to make the transfer so the switch isn't HUGE